src/main/c/Makefile

```diff
@@ -24,6 +24,7 @@ libbwa.$(LIB_EXT): $(JNI_BASE_NAME).o jnibwa.o bwa/libbwa.a
 
 bwa:
 	git clone https://github.com/lh3/bwa && cd bwa && git checkout $(BWA_MEM_COMMIT) && echo '#define BWA_COMMIT "'$(BWA_MEM_COMMIT)'"' > bwa_commit.h
+	sed -i .bak 's/\(LOBJS=.*\)/\1 $$(AOBJS)/g' bwa/Makefile
```

Comment:
Are you still including all the non-library object files with your Makefile sed command?  If so, there is probably an inconsistency there:  two functions with the same name, but different calling sequences, for example.